java/source/dataset.rst

```diff
@@ -25,6 +25,24 @@ Dataset
 
 .. contents::
 
+Arrow Java Dataset offer native functionalities consuming native artifacts such as:
+    - JNI Arrow C++ Dataset: libarrow_dataset_jni (dylib/so):
+        To create C++ natively objects Schema, Dataset, Scanner and export that as a references (long id).
+    - JNI Arrow C Data Interface: libarrow_cdata_jni (dylib/so):
+        To get C++ Recordbacth.
+
+Current supported file format Datasets are:
+    - Parquet
+    - Arrow IPC
+    - ORC
+
+Consider that file format input is an URI it means HDFS/S3 are also supported.
+
+.. note::
+
+    The ScanOptions batchSize argument takes effect only if it is set to a value
+    smaller than the number of rows in the recordbatch.
```

Comment:
I think this is better as API documentation

Reply:
Deleted